public/app/pipeline-orchestrator.js

```
@@ -414,20 +350,87 @@ class PipelineOrchestrator {
+        });
+      }
+
+      // ── Apply IndexedDB model cache patch (ml-worker-fetch-cache.js) ────
+      // If models are cached in IDB, pass Object URLs so the worker skips
+      // re-fetching them from disk. Falls back gracefully if cache is empty.
+      if (typeof window._vipPreloadModels === 'function') {
+        // Fire-and-forget preload — models load while audio context inits.
+        // Only the canonical four models (matching model-loader.js +
+        // models-manifest.json) are preloaded. Any others would diverge from
+        // the SW Cache and double-download under different filenames.
+        window._vipPreloadModels(
```

ESLint

[error] 'window' is not defined.

(no-undef)